### ISSUE TITLE
 Bad history with PSReadLine causes input problems 

### ISSUE THREAD

**Issue by ScatteredRay** [PowerShell/PSReadLine#1744]
# Steps to reproduce

Open AppData/Roaming/Microsoft/Windows/Powershell/PSReadLine/ConsoleHost_history.txt and insert:

```
$AWSTemplate = @{`
    AWSTemplateFormatVersion = "2010-09-09"`
    Description = ""`
    Parameters = @{`
    }`
    Resources = @{`
        Vpc = @{`
            Type = "AWS::EC2::VPC"`
            Properties = @{`
                CidrBlock = "10.0.0.0/24"`
                EnableDnsSupport = $True`
                EnableDnsHostnames = $True`
            }`
        }`
        VpcSubnet = @{`
            Type = "AWS::EC2::Subnet"`
            Properties = @{`
                VpcId = @{ Ref = "Vpc" }`
                CidrBlock = "10.0.0.0/24"`
                MapPublicIpOnLaunch = $True`
            }`
        }`
        SecurityGroup = @{`
            Type = "AWS::EC2::SecurityGroup"`
            Properties = @{`
                GroupDescription = "SecurityGroup"`
                VpcId = @{ Ref = "Vpc" }`
                SecurityGroupIngress = @(`
                    # Ping`
                    @{`
                        CidrIp = "0.0.0.0/0"`
                        IpProtocol = "icmp"`
                        FromPort = -1`
                        ToPort = -1`
                    },`
                    # RDP`
                    @{`
                        CidrIp = "0.0.0.0/0"`
                        IpProtocol = "tcp"`
                        FromPort = 3389`
                        ToPort = 3389`
                    },`
                    # WinRM HTTPS`
                    @{`
                        CidrIp = "0.0.0.0/0"`
                        IpProtocol = "tcp"`
                        FromPort = 5986`
                        ToPort = 5986`
                    }`
                )`
            }`
        }`
        Instance = @{`
            Type = "AWS::EC2::Instance"`
            Properties = @{`
                InstanceType = "t3.large"`
                ImageId = $BaseImageId`
                SecurityGroupIds = @( @{Ref = "SecurityGroup"} )`
                SubnetId = @{ Ref = "VpcSubnet" }`
                KeyName = $KeyPair.KeyName`
            }c`
        }`
    }`
    Outputs = @{`
    }`
}convert
```

now open Windows terminal with PSReadLine and use Ctrl-r to search history and start typing in `AWSTemplate`, See that at some point refresh is messed up.

Hit enter, notice continuing typing leaves extra `bck-i-search' characters in the terminal

# Expected behavior

Windows terminal and PSReadLine to not get in a bad state regardless of what's in the history

# Actual behavior

PSReadLine gets in a bad state

**Comment by ScatteredRay** [comment]
@DHowett mentioned this was an issue in PSReadline here: https://github.com/microsoft/terminal/issues/6926

**Comment by DHowett** [comment]
As a note, this will also reproduce in any console window where the buffer size is locked to the screen size.

**Comment by daxian-dbw** [comment]
This issue was fixed (see #1306). Please upgrade to the 2.0.4 version of PSReadLine from PowerShellGallery. See the [upgrading section](https://github.com/PowerShell/PSReadLine#upgrading) for instructions.

![tabcom](https://user-images.githubusercontent.com/127450/91608673-da626080-e92a-11ea-917d-fa1688510c8f.gif)
